### PR TITLE
Harmonize styles & Add a test that styles are in sync

### LINKED ISSUE #2051: Write an automated test that verifies that the widget styles have the same interface
For the widget styles, we should have a small automated test that takes one of the styles as a reference and make sure that the other styles have the same interface. This test should be run in the CI.

### PR DESCRIPTION
**Pull request by ogoffart**
Closes #2051

### CODE REVIEW

**Comment by ogoffart** [comment]
Of course, tests are currently not in sync so it is failling in CI

File: internal/compiler/tests/consistent_styles.rs
Comment (tronical) [inline internal/compiler/tests/consistent_styles.rs:166]: Nice trick!

**Review by tronical — APPROVED** [review]

**Comment by FloVanGH** [comment]
Nice

**Review by FloVanGH — APPROVED** [review]

**Comment by ogoffart** [comment]
> I noticed that in the failures there's one peculiar failure: The accepted callback is listed as not having the right signature, which looks like it is due to callback being declared as a two-way binding (callback accepted <=> inner.accepted;).

The problem was that it was for a non-existent style called "fluent-base" which is not really a style and doesn't compile on its own as it doesn't find the color file.   and because it had compile error, it would not resolve the aliases.
Fixed by making sure that fluent-base and material-base are not reported as styles

**Comment by ogoffart** [comment]
The question is: how do we proceed now to harmonize the styles?

Should we just make the breaking change now of removing properties from fluent style?  I guess not
Should we remove the extra unwanted properties (eg, from Rectangle) in 1.0?
Should there be a way to make them deprecated?

**Comment by tronical** [comment]
> The question is: how do we proceed now to harmonize the styles?
> 
> Should we just make the breaking change now of removing properties from fluent style? I guess not
> Should we remove the extra unwanted properties (eg, from Rectangle) in 1.0?

I think so, yes.

> Should there be a way to make them deprecated?

I think for 1.0 we should "break" it. But for the next release I think we can either:

1. Don't merge this until after 0.3.4 is out.
2. Add exceptions to this test (either hard-coded or some generic mechanism?) to allow certain extra properties, merge it, and track in a ticket for 1.0.0 to remove all exceptions.

I quite like 2. What do you think?

**Comment by ogoffart** [comment]
I think adding exception is extra work.  
We could just use the result now to fix some low hanging fruit in the difference between material and fluent.
and then remove the properties in 1.0 and push this change

**Comment by tronical** [comment]
> I think adding exception is extra work.
> We could just use the result now to fix some low hanging fruit in the difference between material and fluent.
> and then remove the properties in 1.0 and push this change

Works for me. @FloVanGH what's your opinion?

**Review by FloVanGH — APPROVED** [review]

**Pull request merged**
